src/Spice86.Core/Emulator/Devices/Video/Renderer.cs

```diff
@@ -103,9 +105,19 @@ public void Render(Span<uint> frameBuffer) {
                 for (int scanline = 0; scanline <= maximumScanline; scanline++) {
                     // We latch the destination address here, so that the double-scan lines are drawn on top of each other.
                     int destinationAddressLatch = destinationAddress;
-                    for (int doubleScan = 0; doubleScan < drawLinesPerScanLine; doubleScan++) {
+                    
+                    // VGA aspect ratio correction: calculate lines to draw
+                    // Use actualLineNumber which accounts for scanline within the logical row
+                    int actualLineNumber = lineCounter / (maximumScanline + 1);
+                    // VGA Mode 13h (320x200) needs aspect correction for 5:6 pixel aspect ratio
+                    bool needsAspectCorrection = Width == 320 && actualHeight == 200;
```

Comment:
It has been moved sure.

But I'd rather have you make that signaled by the current emulated video mode received by the video card on SetResolution and updated each time SetResolution event is fired.

Don't use consts inside the renderer, the knowledge itself should be listed in the existing video namespace class that stores static video mode information.

Search the current code base first. The precise class I can't remember right now.